packages/react/src/components/UIShell/__tests__/SideNavButton-test.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright IBM Corp. 2025,
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import SideNavButton from '../SideNavButton';
+
+describe('SideNavButton', () => {
+  it('should apply the custom className to the Button element', () => {
+    render(<SideNavButton className="test">test</SideNavButton>);
+    expect(screen.getByRole('button')).toHaveClass('test');
+  });
+
+  it('should spread extra props on the Button element', () => {
+    render(<SideNavButton data-testid="test">test</SideNavButton>);
+    expect(screen.getByRole('button')).toHaveAttribute('data-testid', 'test');
+  });
+
+  it('should set a `ref` on the Button element', () => {
+    const ref = jest.fn();
+    render(<SideNavButton ref={ref}>test</SideNavButton>);
+    expect(ref).toHaveBeenCalledWith(screen.getByRole('button'));
+  });
+
+  it('should call the `onClick` handler when clicked', async () => {
+    const onClick = jest.fn();
+    const user = userEvent.setup();
+
+    render(<SideNavButton onClick={onClick}>test</SideNavButton>);
+
+    await user.click(screen.getByRole('button'));
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+});
```

Comment:
Should testing be added for these things:

1. `tabIndex` and other accessibility functionlity.
2. The button's children, both static (e.g., text) and dynamic (e.g., a link).
3. `isRail` functionality.
4. Visual testing since one of the key points in https://github.com/carbon-design-system/carbon/issues/18562#issuecomment-2653961793 was broken styles.